pkg/schema/distribution_account.go

```diff
@@ -0,0 +1,45 @@
+package schema
+
+type DistributionAccountType string
+
+const (
+	DistributionAccountTypeStellar DistributionAccountType = "STELLAR"
+	DistributionAccountTypeCircle  DistributionAccountType = "CIRCLE"
+)
+
+type DistributionAccountStatus string
+
+const (
+	DistributionAccountStatusActive                DistributionAccountStatus = "ACTIVE"
+	DistributionAccountStatusPendingUserActivation DistributionAccountStatus = "PENDING_USER_ACTIVATION"
+)
+
+type DistributionAccount struct {
+	ID     string                    `json:"id"`
+	Type   DistributionAccountType   `json:"type"`
+	Status DistributionAccountStatus `json:"status"`
+}
+
+func (da DistributionAccount) IsStellar() bool {
+	return da.Type == DistributionAccountTypeStellar
+}
```

Comment:
Just a suggestion, feel free to ignore. But it seems like it's recurring that we check if an account is of type Stellar, then if it is we fetch the key. So maybe it makes sense to add a method that encapsulates that logic and standardize the error message. E.g. 

```go
func (da DistributionAccount) EnsureStellarAndGetPublicKey() (string, error) {
	if !da.IsStellar() {
		return "", fmt.Errorf("expected distribution account to be a STELLAR account but got %q", da.Type)
	}
	return da.ID, nil
}
```